### PR TITLE
[JOH-50] Implement forgot password rpc

### PR DESCRIPTION
**Pull request by Nitiwat-owen**
- Create email service for sending email via Sendgrid
- Implement forgot password rpc function

# Demo
Screenshot 2567-01-14 at 09 29 45

Screenshot 2567-01-14 at 09 30 11

### CODE REVIEW

File: internal/service/token/token.service.go [hunk 1]
```diff
@@ -167,3 +169,39 @@ func (s *serviceImpl) RemoveRefreshTokenCache(refreshToken string) error {
 
 	return nil
 }
+
+func (s *serviceImpl) CreateResetPasswordToken(userId string) (string, error) {
+	resetPasswordToken := s.CreateRefreshToken()
+	tokenCache := &tokenDto.ResetPasswordTokenCache{
+		UserID: userId,
+	}
+	err := s.resetPasswordTokenCache.SetValue(resetPasswordToken, tokenCache, 900)
```
Comment (bookpanda) [inline internal/service/token/token.service.go]: should we move this ttl to config?

**Review by bookpanda — COMMENTED** [review]
So resetPasswordToken is used to make resetPasswordURL, then how do you plan to to ResetPassword?

File: internal/service/token/token.service.go (hunk 1 above)
Comment (Nitiwat-owen) [reply internal/service/token/token.service.go]: I will retrieve the token from the request , then I will check if the token is exist or not.

Comment (Nitiwat-owen) [reply internal/service/token/token.service.go]: Yes, i think so

Comment (bookpanda) [reply internal/service/token/token.service.go]: I see

**Comment by bookpanda** [comment]
Im figuring out how to review on the phone

**Comment by bookpanda** [comment]
You can merge now, I cant find the review button

**Pull request merged**